flocker/acceptance/test_environment_variables.py

```diff
@@ -166,19 +170,13 @@ def mysql_can_connect():
                     try:
                         return connect(
                             host=host,
-                            port=MYSQL_EXTERNAL_PORT,
+                            port=port,
                             user=user,
                             passwd=passwd,
                             db=db,
                         )
-                    except OperationalError as e:
-                        # PyMySQL doesn't provided a structured way to
-                        # get this.
-                        # https://github.com/PyMySQL/PyMySQL/issues/274
-                        if "Connection refused" in str(e):
-                            return False
-                        else:
-                            raise
+                    except Error:
```

Comment:
Apart from the error mentioned in the JIRA issue, are and "Connection refused" are there any other errors which we know we are OK with? This seems like it is more likely to cause a loop which continues until the timeout is hit if there is a problem.